service/topology_coordinator.cc

```diff
@@ -828,6 +828,50 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
         }
     }
 
+    future<> group0_voter_refresher_fiber() {
+        rtlogger.debug("start group0 members refresh fiber");
+
+        // Skip waiting for the event in the first iteration to start the first voter refresh immediately.
+        //
+        // This is necessary to ensure we update the voters e.g. in case of a topology coordinator change
+        // (the previous topology coordinator might have died in which case we'll need to remove it from
+        // the voters).
+        bool skip_wait = true;
+
+        while (!_as.abort_requested()) {
+            try {
+                // only wait for the event in case we are not retrying the operation
+                if (!skip_wait) {
+                    co_await await_event();
+                    // Introduce a brief delay to potentially batch multiple changes that occur in close succession
+                    co_await seastar::sleep_abortable(std::chrono::milliseconds{100}, _as);
+                }
+                skip_wait = false;
+
+                if (!_feature_service.group0_limited_voters) {
+                    rtlogger.debug("group0 voters refresh fiber iteration skipped because the feature is disabled");
+                    continue;
+                }
+
+                co_await _voter_handler.refresh(_as);
```

Comment:
OK, so we have a refresher fiber that should be able to fix voters if a change during topology operation fails. But does `refresh()` handles it since it pass empty set for added nodes.